File: app/src/main/java/org/amoustakos/boilerplate/util/ConversionUtil.java
package org.amoustakos.boilerplate.util;

import java.text.DecimalFormat;



public final class ConversionUtil {


    /*
     ************** Double conversions *******************
     */

    public static String formatDouble(String value) {
        Double d = getDouble(value, 0D);

        if (d != null)
            return formatDouble(d);
        else
            return value;
    }

    public static Double getDouble(String s, Double defaultNum) {
        if (!StringUtils.isEmptyString(s)) {
            String s2 = s.trim();
            if(StringUtils.isEmptyString(s2))
                return Double.parseDouble(s2);
        }

        return defaultNum;
    }

    /**
     * Formats a double to 2 decimal places.
     */
    public static String formatDouble(double value) {
        return new DecimalFormat("#.##").format(value);
    }


	/*
	 ************** Float conversions *******************
	 */

    public static String formatFloat(String value) {
        Float f = getFloat(value, 0F);

        if (f != null)
            return formatFloat(f);
        else
            return value;
    }
    public static Float getFloat(String s, Float defaultNum) {
        if (!StringUtils.isEmptyString(s)) {
            String s2 = s.trim();
            if(StringUtils.isEmptyString(s2))
                return Float.parseFloat(s2);
        }

        return defaultNum;
    }
    /**
     * Formats a float to 2 decimal places.
     */
    public static String formatFloat(Float value) {
        return new DecimalFormat("#.##").format(value);
    }



	/*
	 ************** Integer conversions *******************
	 */


    public static Integer getInt(String s, Integer defaultNum) {
        if (!StringUtils.isEmptyString(s)) {
            String s2 = s.trim();
            if (StringUtils.isNumeric(s2))
                return Integer.parseInt(s2);
        }
        return defaultNum;
    }



	/*
	 ************** Long conversions *******************
	 */


    public static Long getLong(String s, Long defaultNum) {
        if (!StringUtils.isEmptyString(s)) {
            String s2 = s.trim();
            if (StringUtils.isNumeric(s2))
                return Long.parseLong(s2);
        }
        return defaultNum;
    }



	/*
	 ************** Boolean conversions *******************
	 */


    public static Boolean getBoolean(String s, Boolean defaultBool) {
        if (!StringUtils.isEmptyString(s)) {
            String s2 = s.trim();
            return Boolean.parseBoolean(s2);
        }
        return defaultBool;
    }

    public static Boolean getBooleanFromInt(String s, Boolean defaultBool) {
        if (!StringUtils.isEmptyString(s)) {
            Integer number = getInt(s, null);
            if(number == null)
                return defaultBool;
            else if (number.equals(0))
                return Boolean.FALSE;
            else if (number.equals(1))
                return Boolean.TRUE;
        }
        return defaultBool;
    }


	/*
     ************** Percentage conversions *******************
     */

    public static int getPercent(int value, int max) {
        return (value * 100) / max;
    }
}

File: app/src/main/java/org/amoustakos/boilerplate/util/text/ConversionUtil.kt
package org.amoustakos.boilerplate.util.text

import java.text.DecimalFormat


object ConversionUtil {

    // =========================================================================================
    // Double conversions
    // =========================================================================================

    fun formatDouble(value: String): String {
        val d = getDouble(value, 0.0)

        return if (d != null)
            formatDouble(d)
        else
            value
    }

    fun getDouble(s: String, defaultNum: Double?): Double? {
        if (!StringUtils.isEmptyString(s)) {
            val s2 = s.trim { it <= ' ' }
            if (StringUtils.isEmptyString(s2))
                return java.lang.Double.parseDouble(s2)
        }

        return defaultNum
    }

    /**
     * Formats a double to 2 decimal places.
     */
    fun formatDouble(value: Double): String {
        return DecimalFormat("#.##").format(value)
    }


    // =========================================================================================
    // Float conversions
    // =========================================================================================

    fun formatFloat(value: String): String {
        val f = getFloat(value, 0f)

        return if (f != null)
            formatFloat(f)
        else
            value
    }

    fun getFloat(s: String, defaultNum: Float?): Float? {
        if (!StringUtils.isEmptyString(s)) {
            val s2 = s.trim { it <= ' ' }
            if (StringUtils.isEmptyString(s2))
                return java.lang.Float.parseFloat(s2)
        }

        return defaultNum
    }

    /**
     * Formats a float to 2 decimal places.
     */
    fun formatFloat(value: Float?): String {
        return DecimalFormat("#.##").format(value)
    }


    // =========================================================================================
    // Integer conversions
    // =========================================================================================


    fun getInt(s: String, defaultNum: Int?): Int? {
        if (!StringUtils.isEmptyString(s)) {
            val s2 = s.trim { it <= ' ' }
            if (StringUtils.isNumeric(s2))
                return Integer.parseInt(s2)
        }
        return defaultNum
    }


    // =========================================================================================
    // Long conversions
    // =========================================================================================


    fun getLong(s: String, defaultNum: Long?): Long? {
        if (!StringUtils.isEmptyString(s)) {
            val s2 = s.trim { it <= ' ' }
            if (StringUtils.isNumeric(s2))
                return java.lang.Long.parseLong(s2)
        }
        return defaultNum
    }


    // =========================================================================================
    // Boolean conversions
    // =========================================================================================


    fun getBoolean(s: String, defaultBool: Boolean?): Boolean? {
        if (!StringUtils.isEmptyString(s)) {
            val s2 = s.trim { it <= ' ' }
            return java.lang.Boolean.parseBoolean(s2)
        }
        return defaultBool
    }

    fun getBooleanFromInt(s: String, defaultBool: Boolean?): Boolean? {
        if (!StringUtils.isEmptyString(s)) {
            val number = getInt(s, null)
            if (number == null)
                return defaultBool
            else if (number == 0)
                return java.lang.Boolean.FALSE
            else if (number == 1)
                return java.lang.Boolean.TRUE
        }
        return defaultBool
    }


    // =========================================================================================
    // Percentage conversions
    // =========================================================================================

    fun getPercent(value: Int, max: Int): Int {
        return value * 100 / max
    }
}
